Documentation/src/case-studies/calendar.md

```diff
@@ -0,0 +1,350 @@
+
+
+This is actually pretty straight forward. What do we want? We want to format a single month, and how would a single month look like? So the result will be a sequence of strings, each string representing one line of the layout.
+
+```
+        April       
+                1  2
+ 3  4  5  6  7  8  9
+10 11 12 13 14 15 16
+17 18 19 20 21 22 23
+24 25 26 27 28 29 30
+```
+
+First, we decompose the problem once more: we start with one line where we print the month, then a few lines which represent a week and we also have an empty line at the end, but that is just layout sugar.
+
+We need a few helper functions again: the most important step is grouping the days by week, and then we pass each week to a FormatWeek function.
+
+```cs
+private static IEnumerable<string> LayoutMonth(IEnumerable<DateOnly> month)
+{
+    yield return CenteredMonthName(month);
+
+    foreach (var week in month.AdjacentGroupBy(GetWeekOfYear).Select(FormatWeek))
+    {
+        yield return week;
+    }
+
+    yield return $"{string.Empty,WidthOfAWeek}";
+} 
+```
+
+This is a fine way to do this, it is very easy to read, but the foreach is really annoying. We already have an `IEnumerable<T>`, we just want to add it between the first and the last line. But with yield you are often limited to very procedural constructs.
+
+Often you can avoid that, for simple cases we have Sequence.Return, Concat and other helpers, in this case though the nicest way is probaly creating an ImmutableList, because the syntax allows to combine ranges and single items elegantly.
+
+```cs
+private static IEnumerable<string> LayoutMonth(IEnumerable<DateOnly> month)
+    => ImmutableList<string>.Empty
+        .Add(CenteredMonthName(month))
+        .AddRange(FormatWeeks(month))
+        .Add($"{string.Empty,WidthOfAWeek}");
+```
+
+Now we dive into the helper functions. First we take a look at the name of the month. The only noteworthy detail is the very functional mindest seen in the solution to the centering problem. It uses a pattern match to fill in the missing spaces: it is not very efficent, but easy to understand. The recursion will be very short because our lines are only 21 characters wide.
+
+
+```cs
+private static string CenteredMonthName(IEnumerable<DateOnly> month)
+    => month
+        .First()
+        .ToString(MonthNameFormat)
+        .Center(WidthOfAWeek );
+
+
+internal static class StringExtensions
+{
+    public static string Center(this string text, int width)
+        => (width - text.Length) switch
+        {
+            0 => text,
+            1 => $" {text}",
+            _ => Center($" {text} ", width),
+        };
+}
+```
+
+We have already seen the heart of FormatWeeks in the yield solution, but now it is a separate function. FormatWeeks again needs 2 very simple helper functions, the first one Projects the Week of the year, the other one will format a sequence of days.
```

Comment:
FormatWeeks should be in single ticks

Reply:
Fixed in 239b3a73af300f846afba75f4a992650b6fcb41b